### ISSUE TITLE
Remove usage of primitive wrapper constructors

### ISSUE THREAD

**Issue by Thihup** [jakartaee/platform-tck#646]
**Is your feature request related to a problem? Please describe.**
Starting with JDK 16, the primitive wrappers are deprecated for removal, so it means that probably shortly they will remove these constructors. [1] 

**Describe the solution you'd like**
Replace the usage of `new Integer`, `new Character`, `new Double`, etc with `Integer.valueOf`, `Character.valueOf`, etc.

**Describe alternatives you've considered**
If we do not migrate remove the usage of these constructors, the implementations will be able to certify themselves with the last JDK version that supports these constructors (JDK 16, JDK 17).
As this project has a lot of matches, doing it in steps would be a great way to handle this big change.

**Additional context**
[1] https://openjdk.java.net/jeps/390

**Comment by scottmarlow** [comment]
If I understand correctly, this issue can only be addressed for a Jakarta EE release that bases the Java SE version on JDK 16 or greater.

**Comment by Thihup** [comment]
@scottmarlow 
I don't think so. What I propose here is change `new Double(5)` for `Double.valueOf(5)`.
These factories methods are from Java 5 IIRC.
